nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/deck_guns_construction.dm

```diff
@@ -0,0 +1,456 @@
+		return TRUE
+
+	else if(istype(W, /obj/item/stack/cable_coil) && (state == BS_IGNITERS_SECURED))
+		var/obj/item/stack/cable_coil/S = W
+		if(S.get_amount() < num_cables)
+			to_chat(user, "<span class='warning'>You need four pieces of [S] to wire the NAC!</span>")
+			return
+		if(!do_after(user, 2 SECONDS, target=src))
+			return
+		if(S.get_amount() < num_cables)
+			to_chat(user, "<span class='warning'>You need four pieces of [S] to wire the NAC!</span>")
+			return
+		S.use(num_cables)
+		to_chat(user, "<span class='notice'>You wire up the electronics.</span>")
+		state = BS_WIRED
+		return TRUE
+
+	else if((istype(W, /obj/item/stack/sheet/plasteel)) && (state == BS_WIRES_SOLDERED))
+		var/obj/item/stack/sheet/plasteel/S = W
+		if(S.get_amount() < num_sheets_casing)
+			to_chat(user, "<span class='warning'>You need four sheets of [S] to finish the [src]!</span>")
+			return
+		if(!do_after(user, 2 SECONDS, target=src))
+			return
+		if(S.get_amount() < num_sheets_casing)
+			to_chat(user, "<span class='warning'>You need four sheets of [S] to finish the [src]!</span>")
+			return
+		S.use(num_sheets_casing)
+		to_chat(user, "<span class='notice'>You add the outer casing to the assembly.</span>")
+		state = BS_CASING_ADDED
+		return TRUE
+
+	else if((istype(W, /obj/item/ship_weapon/parts/mac_barrel)) && (state == BS_CASING_ADDED))
+		W.forceMove(src)
+		barrels_added++
+		to_chat(user, "<span class='notice'>You add the [W] to the [src].</span>")
+		if(barrels_added >= max_barrels)
+			state = BS_BARRELS_PLACED
+		return TRUE
+
+	else if((istype(W, /obj/item/ship_weapon/parts/loading_tray)) && (state == BS_BARRELS_SECURED))
+		if(!do_after(user, 2 SECONDS, target=src))
+			return
+		W.forceMove(src)
+		to_chat(user, "<span class='notice'>You slide the loading tray into place.</span>")
+		if(ispath(text2path("/obj/machinery/ship_weapon/deck_turret/_[max_barrels]")))
+			output_path = text2path("/obj/machinery/ship_weapon/deck_turret/_[max_barrels]")
+		var/obj/machinery/ship_weapon/built = new output_path(loc)
+		built.setDir(dir)
+		built.setAnchored(anchored)
+		built.on_construction()
+		built.max_ammo = max_barrels
+
+		for(var/obj/O in built.component_parts)
+			qdel(O)
+			stoplag()
+		transfer_fingerprints_to(built)
+		built.component_parts = list()
+		for(var/obj/O in src)
+			O.moveToNullspace()
+			built.component_parts += O
+			stoplag()
+		qdel(src)
+		return TRUE
+
+	..()
+	
+/obj/structure/ship_weapon/artillery_frame/attack_robot(mob/user)
+	. = ..()
+	attack_hand(user)
+
+/obj/structure/ship_weapon/artillery_frame/attack_hand(mob/user)
+	. = ..()
+	if(state == BS_IGNITERS_PLACED)
+		if(!do_after(user, 2 SECONDS, target=src))
+			return
+		to_chat(user, "<span class='notice'>You remove the igniter[(max_barrels != 1) ? "s" : ""].</span>")
+		var/obj/item/assembly/igniter/C = (locate(/obj/item/assembly/igniter) in src)
+		while(C)
+			C.forceMove(user.loc)
+			C = (locate(/obj/item/assembly/igniter) in src)
+			stoplag()
```

Comment:
```suggestion
		for(var/obj/item/assembly/igniter/C in src)
			C.forceMove(user.loc)
```

Reply:
I'm gonna do this everywhere it's applicable.